### PR TITLE
bump controller-runtime version in odh-notebook-controller

### PR DESCRIPTION
**Pull request by rkpattnaik780**
## Description

The PR bumps version of the golang dependency `sigs.k8s.io/controller-runtime` from `0.11.0` to `0.15.0` for the components `notebook-controller` and `odh-notebook-controller`.
## How Has This Been Tested?

Create a workbench in a data science project after updating the `odh-notebook-controller-manager` to use the image `quay.io/opendatahub/odh-notebook-controller:pr-241`.

## Merge criteria:

- [X] The commits are squashed in a cohesive manner and have meaningful messages.
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: components/notebook-controller/go.sum [hunk 1]
```diff
@@ -672,52 +628,31 @@ github.com/cncf/xds/go v0.0.0-20211011173535-cb28da3451f1/go.mod h1:eXthEFrGJvWH
...
 github.com/creack/pty v1.1.9/go.mod h1:oKZEueFk5CKHvIhNR5MUki03XCEU+Q6VDXinZuGJ33E=
-github.com/creack/pty v1.1.11/go.mod h1:oKZEueFk5CKHvIhNR5MUki03XCEU+Q6VDXinZuGJ33E=
 github.com/davecgh/go-spew v1.1.0/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
 github.com/davecgh/go-spew v1.1.1 h1:vj9j/u1bqnvCEfJOwUhtlOARqs3+rkHYY13jYWTU97c=
 github.com/davecgh/go-spew v1.1.1/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
-github.com/dgrijalva/jwt-go v3.2.0+incompatible/go.mod h1:E3ru+11k8xSBh+hMPgOLZmtrrCbhqsmaPHjLKYnJCaQ=
-github.com/dgryski/go-sip13 v0.0.0-20181026042036-e10d5fee7954/go.mod h1:vAd38F8PWV+bWy6jNmig1y/TA+kYO4g3RSRF0IAv0no=
 github.com/docopt/docopt-go v0.0.0-20180111231733-ee0de3bc6815/go.mod h1:WwZ+bS3ebgob9U8Nd0kOddGdZWjyMGR8Wziv+TBNwSE=
 github.com/dustin/go-humanize v1.0.0/go.mod h1:HtrtbFcZ19U5GC7JDqmcUSB87Iq5E25KnS6fMYU6eOk=
-github.com/elazarl/goproxy v0.0.0-20180725130230-947c36da3153/go.mod h1:/Zj4wYkgs4iZTTu3o/KG3Itv/qCCa8VVMlb3i9OVuzc=
```
Comment (rkpattnaik780) [inline components/notebook-controller/go.sum:695]: The [CVE-2023-37788](https://www.cve.org/CVERecord?id=CVE-2023-37788) was introduced through this dependency.

**Review by harshad16 — COMMENTED** [review]
Applying the changes to the cluster break the notebook-controller.
```
E0130 06:25:12.641300 1 runtime.go:79] Observed a panic: &runtime.TypeAssertionError{_interface:(*runtime._type)(0x179ac80), concrete:(*runtime._type)(0x17f2880), asserted:(*runtime._type)(0x17f12c0), missingMethod:""} (interface conversion: client.Object is *v1.Pod, not *v1.Event)
goroutine 72 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x164b8c0?, 0xc000902750})
/go/pkg/mod/k8s.io/apimachinery@v0.27.2/pkg/util/runtime/runtime.go:75 +0x99
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x0?})
/go/pkg/mod/k8s.io/apimachinery@v0.27.2/pkg/util/runtime/runtime.go:49 +0x75
```

Need to go over the code a little bit closely, will provide second review sooner

**Comment by rkpattnaik780** [comment]
> Applying the changes to the cluster break the notebook-controller.
> 
> ```
> E0130 06:25:12.641300 1 runtime.go:79] Observed a panic: &runtime.TypeAssertionError{_interface:(*runtime._type)(0x179ac80), concrete:(*runtime._type)(0x17f2880), asserted:(*runtime._type)(0x17f12c0), missingMethod:""} (interface conversion: client.Object is *v1.Pod, not *v1.Event)
> goroutine 72 [running]:
> k8s.io/apimachinery/pkg/util/runtime.logPanic({0x164b8c0?, 0xc000902750})
> /go/pkg/mod/k8s.io/apimachinery@v0.27.2/pkg/util/runtime/runtime.go:75 +0x99
> k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x0?})
> /go/pkg/mod/k8s.io/apimachinery@v0.27.2/pkg/util/runtime/runtime.go:49 +0x75
> ```
> 
> Need to go over the code a little bit closely, will provide second review sooner

Thanks for the review. This might have been caused by the following change:
`https://github.com/opendatahub-io/kubeflow/pull/241/files#diff-a436986c5e93996f9035eee6798d064ab8659957e8443eb44b3a4d3db289bde4R741`

Have updated the PR.

File: components/notebook-controller/Makefile [hunk 2]
```diff
@@ -135,20 +135,25 @@ deploy: docker-build docker-push manifests kustomize ## Deploy controller to the
...
 kustomize: ## Download kustomize locally if necessary.
 	$(call go-get-tool,$(KUSTOMIZE),sigs.k8s.io/kustomize/v3/cmd/kustomize@v3.2.0)
 
-ENVTEST = $(shell pwd)/bin/setup-envtest
+ENVTEST = $(LOCALBIN)/setup-envtest
 .PHONY: envtest
-envtest: ## Download envtest-setup locally if necessary.
-	$(call go-get-tool,$(ENVTEST),sigs.k8s.io/controller-runtime/tools/setup-envtest@v0.0.0-20230216140739-c98506dc3b8e)
+envtest: ## Download setup-envtest locally if necessary.
+	GOBIN=$(LOCALBIN) go install sigs.k8s.io/controller-runtime/tools/setup-envtest@v0.0.0-20230216140739-c98506dc3b8e
```
Comment (harshad16) [inline components/notebook-controller/Makefile:156]: Are these changes needed ? why is it needed

**Review by harshad16 — CHANGES_REQUESTED** [review]
lgtm, tested it works well
just one more question:

File: components/notebook-controller/Makefile (hunk 2 above)
Comment (harshad16) [reply components/notebook-controller/Makefile:156]: Never seems like i found the answer:
https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/opendatahub-io_kubeflow/247/pull-ci-opendatahub-io-kubeflow-v1.7-branch-kf-notebook-controller-unit/1753521090027589632
the ci fails to locate the resultant controller-gen.

ack, this fix the ci test.

**Review by harshad16 — APPROVED** [review]
/lgtm
/approve

Tested in a setup cluster, it seems to be running as required.

**Pull request merged**